src/diffusion_for_multi_scale_molecular_dynamics/models/axl_diffusion_lightning_model.py

```diff
@@ -180,73 +254,164 @@ def _generic_step(
         )
-
-        unreduced_loss = self.loss_calculator.calculate_unreduced_loss(
-            predicted_normalized_scores,
-            target_normalized_conditional_scores,
+        # this output is expected to be an AXL object
+        # X score network output: an estimate of the sigma normalized score for the coordinates,
+        # A score network output: an unnormalized estimate of p(a_0 | a_t) for the atom types
+        # TODO something for the lattice
+
+        unreduced_loss_coordinates = self.loss_calculator.X.calculate_unreduced_loss(
+            model_predictions.X,
+            target_coordinates_normalized_conditional_scores,
             sigmas,
         )
-        loss = torch.mean(unreduced_loss)
+
+        unreduced_loss_atom_types = self.loss_calculator.A.calculate_unreduced_loss(
+            predicted_unnormalized_probabilities=model_predictions.A,
+            one_hot_real_atom_types=a0_onehot,
+            one_hot_noisy_atom_types=at_onehot,
+            time_indices=noise_sample.indices,
+            q_matrices=q_matrices,
+            q_bar_matrices=q_bar_matrices,
+            q_bar_tm1_matrices=q_bar_tm1_matrices,
+        )
+
+        # TODO placeholder - returns zero
+        unreduced_loss_lattice = self.loss_calculator.L.calculate_unreduced_loss(
+            model_predictions.L
+        )
+
+        # TODO consider having weights in front of each component
+        aggregated_loss = (
+            unreduced_loss_coordinates.mean(
+                dim=-1
+            )  # batch, num_atoms, spatial_dimension
+            + unreduced_loss_lattice
+            + unreduced_loss_atom_types.mean(dim=-1)  # batch, num_atoms, num_atom_types
+        )
+
+        loss = torch.mean(aggregated_loss)
+
+        unreduced_loss = AXL(
+            A=unreduced_loss_atom_types.detach(),
+            X=unreduced_loss_coordinates.detach(),
+            L=torch.zeros_like(
+                unreduced_loss_coordinates
+            ).detach(),  # TODO use unreduced_loss_lattice.detach(),
+        )
+
+        model_predictions_detached = AXL(
+            A=model_predictions.A.detach(),
+            X=model_predictions.X.detach(),
+            L=model_predictions.L.detach(),
+        )
 
         output = dict(
-            unreduced_loss=unreduced_loss.detach(),
+            unreduced_loss=unreduced_loss,
             loss=loss,
             sigmas=sigmas,
-            predicted_normalized_scores=predicted_normalized_scores.detach(),
-            target_normalized_conditional_scores=target_normalized_conditional_scores,
+            model_predictions=model_predictions_detached,
+            target_coordinates_normalized_conditional_scores=target_coordinates_normalized_conditional_scores,
         )
-        output[RELATIVE_COORDINATES] = x0
-        output[NOISY_RELATIVE_COORDINATES] = augmented_batch[NOISY_RELATIVE_COORDINATES]
+        output[ORIGINAL_AXL] = original_sample
+        output[NOISY_AXL] = NOISY_AXL
```

Comment:
`NOISY_AXL` is a string. It's ok that this is used as a key in the `output` dictionary, bu I think the value be `noisy_sample`.